pkg/build/registry/buildconfig/webhook.go

```diff
@@ -72,9 +73,13 @@ func (w *WebHook) ServeHTTP(writer http.ResponseWriter, req *http.Request, ctx k
 		Revision:    revision,
 		Env:         envvars,
 	}
-	if _, err := w.instantiator.Instantiate(config.Namespace, request); err != nil {
+	newBuild, err := w.instantiator.Instantiate(config.Namespace, request)
+	if err != nil {
 		return errors.NewInternalError(fmt.Errorf("could not generate a build: %v", err))
 	}
+	// Send back the build name so that the client can alert the user.
+	newBuildJson, _ := json.Marshal(newBuild)
```

Comment:
> I was trying to avoid an api that potentially returns two different types. The biggest challenge here is it's not exactly a "failure" case. The build was launched, but we can't return you the build object.

That's already a thing.  `curl https://localhost:8443/api/v1/namespaces/foo` if "foo" doesn't exist returns a `kind: Status`.  `curl -XDELETE https://localhost:8443/api/v1/namespaces/foo/something/bar` has the option of returning either the resource you deleted or a status object on a 200 return.